### PR TITLE
Display total number of courses in course listing headers

### LINKED ISSUE #1815: Add serial number and total number of courses to Course Administration page
URL is `admin/courses`

Add a serial number to the table showing the list of courses.
Display the total number of courses at the top.

### PR DESCRIPTION
**Pull request by fonglh**
Show total number of courses in the header for both system course listing and instance course listing.

When search is used, show the number of search results instead.

References #1815. Partial fix, serial number yet to be discussed.

### CODE REVIEW

File: app/views/system/admin/courses/index.html.slim [hunk 1]
```diff
@@ -1,10 +1,11 @@
-= page_header
+= page_header t('.header', count: @courses.count)
```
Comment (allenwq) [inline app/views/system/admin/courses/index.html.slim]: Page header defaults to a translation, remove or reuse that.

**Comment by allenwq** [comment]
@fonglh  The course counter will start from 0 again in the second page.

**Comment by allenwq** [comment]
Instead of the head. I think we should provide a summary section similar to V1, and this can be shared by both system and instance admin.

File: app/views/system/admin/courses/index.html.slim (hunk 1 above)
Comment (fonglh) [inline app/views/system/admin/courses/index.html.slim]: Yes, the `.header` translation is reused.

**Comment by fonglh** [comment]
If we do the summary section, the pagination will still be problematic right?

**Comment by allenwq** [comment]
Summary is for displaying the total course number and maybe active courses later. 

The SN one need a different solution.

**Comment by weiqingtoh** [comment]
@fonglh can try calculated attributes scoped to an instance.

File: app/views/system/admin/courses/index.html.slim [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+/ Show total number of courses when search field is blank (nil or empty string)
+- if params['search'].blank?
+  = page_header t('.header_with_count', count: Course.unscoped.all.count)
+- else
+  = page_header
 
 = render partial: 'layouts/search_form', locals: { url: admin_courses_path, placeholder: t('.search_placeholder') }
 
+- unless params['search'].blank?
+  h4 = t('.search_count', count: @courses.map(&:title).count)
```
Comment (allenwq) [inline app/views/system/admin/courses/index.html.slim]: Will `@courses.size` work ?

Comment (fonglh) [inline app/views/system/admin/courses/index.html.slim]: Yes it does, updated.

File: app/views/system/admin/instance/courses/index.html.slim [hunk 3]
```diff
@@ -1,7 +1,14 @@
-= page_header
+/ Show  number of courses in instance when search field is blank (nil or empty string)
+- if params['search'].blank?
+  = page_header t('.header_with_count', count: @courses.map(&:title).count)
```
Comment (allenwq) [inline app/views/system/admin/instance/courses/index.html.slim]: @fonglh You forgot to update here

**Pull request merged**